platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/problems/ConfigurationCacheProblems.kt

```diff
@@ -80,7 +85,10 @@ class ConfigurationCacheProblems(
     val failureFactory: FailureFactory,
 
     private
-    val buildNameProvider: BuildNameProvider
+    val buildNameProvider: BuildNameProvider,
+
+    private val
+    degradationController: DefaultConfigurationCacheDegradationController
```

Comment:
🤔 This dependency smells odd

Reply:
> that `CCP` cares about degradation reasons at all?

Yes, that it needs to interact with degradation controller.

Reply:
That is similar to how it needs to track incompatible tasks, right? The main difference is that degradation requests are managed by `CCP` itself, whereas degradation is encapsulated by a separate class (which `CCP` is then a client of).